app/controllers/ncr/work_orders_controller.rb

```diff
@@ -21,18 +21,32 @@ def edit
     end
 
     def update
+      @model_instance.assign_attributes(permitted_params)
       @model_instance.modifier = current_user
 
-      super
+      if errors.empty?
+        update_or_notify_of_no_changes
+        redirect_to proposal_path(@model_instance.proposal)
+      else
+        flash[:error] = errors
+        render :edit
+      end
+    end
 
-      if @model_changing
+    protected
+
+    def update_or_notify_of_no_changes
+      if attribute_changes?
+        ProposalUpdateRecorder.new(@model_instance).run
+        @model_instance.save
         @model_instance.setup_approvals_and_observers
         @model_instance.email_approvers
+        flash[:success] = "Successfully modified!"
+      else
+        flash[:error] = "No changes were made to the request"
       end
     end
```

Comment:
Not sure that I agree with the changes in this file, since you're copying in the functionality from the `UseCaseController`... Would you mind if we decouple the `ProposalUpdateRecorder` change from the removal of the `super` call in the `update` action above and do them in separate PRs?

/cc https://github.com/18F/C2/pull/756